azext_edge/edge/providers/rpsaas/adr/base.py

```diff
@@ -0,0 +1,210 @@
+)
+
+from .user_strings import (
+    CUSTOM_LOCATION_DOES_NOT_EXIST_ERROR,
+    CUSTOM_LOCATION_NOT_FOUND_ERROR,
+    CLUSTER_NOT_FOUND_ERROR,
+    CLUSTER_OFFLINE,
+    MISSING_CLUSTER_CUSTOM_LOCATION_ERROR,
+    MISSING_EXTENSION_ERROR,
+    MULTIPLE_CUSTOM_LOCATIONS_ERROR,
+    MULTIPLE_POSSIBLE_ITEMS_ERROR
+)
+from ..base_provider import RPSaaSBaseProvider
+from ....util import build_query
+from ....common import ResourceTypeMapping
+
+logger = get_logger(__name__)
+API_VERSION = "2023-11-01-preview"
+
+
+# @vilit - prob generalize this even more
+class ADRBaseProvider(RPSaaSBaseProvider):
+    def __init__(
+        self, cmd, resource_type: str
+    ):
+        super(ADRBaseProvider, self).__init__(
+            cmd=cmd,
+            api_version=API_VERSION,
+            resource_type=resource_type,
+            required_extension="microsoft.deviceregistry.assets"
+        )
+
+    def delete(
+        self,
+        resource_name: str,
+        resource_group_name: str
+    ):
+        self.show_and_check(
+            resource_name=resource_name,
+            resource_group_name=resource_group_name
+        )
+        return self.resource_client.resources.begin_delete(
+            resource_group_name=resource_group_name,
+            resource_provider_namespace=self.resource_type,
+            parent_resource_path="",
+            resource_type="",
+            resource_name=resource_name,
+            api_version=self.api_version
+        )
+
+    def show_and_check(
+        self,
+        resource_name: str,
+        resource_group_name: str
+    ) -> Dict[str, Any]:
+        result = self.show(resource_name, resource_group_name)
+        self._check_cluster_connectivity(
+            result["extendedLocation"]["name"]
+        )
+        return result
+
+    def _check_cluster_connectivity(
+        self,
+        custom_location_id: str,
+    ):
+        query = f'| where id =~ "{custom_location_id}"'
+        custom_location_query_result = build_query(
+            self.cmd,
+            custom_query=query,
+            type=ResourceTypeMapping.custom_location.value
+        )
+        if len(custom_location_query_result) == 0:
+            logger.warning(CUSTOM_LOCATION_NOT_FOUND_ERROR.format(custom_location_id))
+            return
+        query = f'| where id =~ "{custom_location_query_result[0]["properties"]["hostResourceId"]}"'
+        cluster_query_result = build_query(
+            self.cmd,
+            custom_query=query,
+            type=ResourceTypeMapping.connected_cluster.value
+        )
+        # TODO: add option to fail on these
+        if len(cluster_query_result) == 0:
+            logger.warning(CLUSTER_NOT_FOUND_ERROR.format(custom_location_id))
+            return
+        cluster = cluster_query_result[0]
+        if cluster["properties"]["connectivityStatus"].lower() != "connected":
+            logger.warning(CLUSTER_OFFLINE.format(cluster["name"]))
+
+    def _check_cluster_and_custom_location(
```

Comment:
Not sure why this method is prefixed as private, when child class AssetEndpointProfileProvider calls it directly.

Same with `_check_cluster_connectivity`.

Reply:
I thought child classes calling private methods is stylistically fine? Though in Java world that isn't so I will fix.